src/nvme/fabrics.c

```diff
@@ -780,37 +780,20 @@ nvme_ctrl_t nvmf_connect_disc_entry(nvme_host_t h,
 	return NULL;
 }
 
-static int nvme_discovery_log(int fd, __u32 len, struct nvmf_discovery_log *log, bool rae)
-{
-	struct nvme_get_log_args args = {
-		.args_size = sizeof(args),
-		.fd = fd,
-		.nsid = NVME_NSID_NONE,
-		.lsp = NVME_LOG_LSP_NONE,
-		.lsi = NVME_LOG_LSI_NONE,
-		.uuidx = NVME_UUID_NONE,
-		.timeout = NVME_DEFAULT_IOCTL_TIMEOUT,
-		.result = NULL,
-		.lid = NVME_LOG_LID_DISCOVER,
-		.log = log,
-		.len = len,
-		.csi = NVME_CSI_NVM,
-		.rae = rae,
-		.ot = false,
-	};
-
-	return nvme_get_log_page(fd, 4096, &args);
-}
-
-int nvmf_get_discovery_log(nvme_ctrl_t c, struct nvmf_discovery_log **logp,
-			   int max_retries)
+int nvmf_get_discovery_log2(nvme_ctrl_t c,
+			    struct nvme_get_log_args *args,
+			    struct nvmf_discovery_log **logp,
+			    int max_retries)
```

Comment:
Maybe we should just introduce a variant (and keep it even for an 2.x update where we could change the arguments of the original function) with `struct nvme_get_log_args` argument and have a better name (though I don't have a good idea here `nvmf_get_discover_log_extended`?